server/settings_test.go

```diff
@@ -559,16 +559,52 @@ func TestSettings(t *testing.T) {
 				})
 			})
 
-			t.Run("AlertManager", func(t *testing.T) {
+			t.Run("AlertingRules", func(t *testing.T) {
+				type annotations struct {
+					Summary string `json:"summary"`
+				}
+				type labels struct {
+					Alertname string `json:"alertname"`
+					Severity  string `json:"severity"`
+				}
+				type alerts struct {
+					ActiveAt    time.Time   `json:"activeAt"`
+					Annotations annotations `json:"annotations"`
+					Labels      labels      `json:"labels"`
+					State       string      `json:"state"`
+					Value       string      `json:"value"`
+				}
+				type alertingRules struct {
+					Alerts      []alerts    `json:"alerts,omitempty"`
+					Annotations annotations `json:"annotations,omitempty"`
+					Duration    int         `json:"duration,omitempty"`
+					Health      string      `json:"health"`
+					Labels      labels      `json:"labels,omitempty"`
+					Name        string      `json:"name"`
+					Query       string      `json:"query"`
+					Type        string      `json:"type"`
+				}
+				type alertGroup struct {
+					Rules    []alertingRules `json:"rules"`
+					File     string          `json:"file"`
+					Interval int             `json:"interval"`
+					Name     string          `json:"name"`
+				}
+				type prometheusResponse struct {
+					Data struct {
+						Groups []alertGroup `json:"groups"`
+					} `json:"data"`
+					Status string `json:"status"`
+				}
```

Comment:
All those types are already defined together with Prometheus API client: https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api.go

Reply:
I think you missed a part about "Prometheus API client". Why we don't use it
https://github.com/prometheus/client_golang/blob/03575cad4e55adb37a5396afa7cc0eec43526740/api/prometheus/v1/api.go#L793